test/library/keycloakidp.go

```diff
@@ -0,0 +1,374 @@
+package library
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+)
+
+func AddKeycloakIDP(
+	t *testing.T,
+	kubeconfig *rest.Config,
+) (idpURL, idpName string, cleanups []func()) {
+	kubeClients, err := kubernetes.NewForConfig(kubeconfig)
+	require.NoError(t, err)
+
+	routeClient, err := routev1client.NewForConfig(kubeconfig)
+	require.NoError(t, err)
+
+	configClient, err := configv1client.NewForConfig(kubeconfig)
+	require.NoError(t, err)
+
+	// configclients, err := configv1client.NewForConfig(kubeconfig)
+	// require.NoError(t, err)
+
+	nsName, keycloakHost, cleanup := deployPod(t, kubeClients, routeClient,
+		"keycloak",
+		"docker.io/jboss/keycloak:latest",
```

Comment:
We prefer to use Quay: https://quay.io/repository/keycloak/keycloak

Reply:
Thanks!